assets/dev/js/editor/elements/views/base.js

```diff
@@ -619,16 +625,150 @@ BaseElementView = BaseContainer.extend( {
 				return;
 			}
 
-			if ( ! isContentChanged ) {
+			if ( shouldRenderUI && ! isContentChanged ) {
 				this.renderUI();
+
 				return;
 			}
 		}
 
-		// Re-render the template
 		this.renderHTML();
 	},
 
+	/**
+	 * Function linkDataBindings().
+	 *
+	 * Link data to allow partial render, instead of full re-render
+	 *
+	 * How to use?
+	 *  If the element which should be rendered for a setting key is known in advance, it's possible to add the following attributes to the element to avoid full re-render:
+	 *  Example for repeater item:
+	 * 'data-binding-type': 'repeater-item',               // Type of binding (to know how to behave).
+	 * 'data-binding-setting': 'tab_title',                // Setting key that effect the binding.
+	 * 'data-binding-index': tabCount,                     // Index is required for repeater items.
+	 *
+	 * Example for content:
+	 * 'data-binding-type': 'content',                     // Type of binding.
+	 * 'data-binding-setting': 'testimonial_content',      // Setting change to capture, the value will replace the link.
+	 *
+	 * By adding the following example attributes inside the widget the element innerHTML will be linked to the 'testimonial_content' setting value.
+	 *
+	 * Current Limitation:
+	 * Not working with dynamics, will required full re-render.
+	 */
+	linkDataBindings() {
+		/**
+		 * @type {Array.<DataBinding>}
+		 */
+		this.dataBindings = [];
+
+		const id = this.$el.data( 'id' );
+
+		if ( ! id ) {
+			return;
+		}
+
+		const $dataBinding = this.$el.find( '[data-binding-type]' );
+
+		if ( ! $dataBinding.length ) {
+			return;
+		}
+
+		const self = this;
+
+		$dataBinding.filter( function() {
+			const $current = jQuery( this );
+
+			// To support nested data-binding bypass nested data-binding that are not part of the current.
+			if ( $current.closest( '.elementor-element' ).data( 'id' ) === id ) {
+				if ( this.dataset.bindingType ) {
+					self.dataBindings.push( {
+						el: this,
+						dataset: this.dataset,
+					} );
+				}
+			}
+		} );
```

Comment:
```suggestion
		$dataBinding.forEach( ($current) => {
			// To support nested data-binding bypass nested data-binding that are not part of the current.
			if ( 
				$current.closest( '.elementor-element' ).data( 'id' ) === id &&
				this.dataset.bindingType
			) {
				this.dataBindings.push( {
					el: this,
					dataset: this.dataset,
				} );
			}
		} );
```
Maybe can be simplified

Reply:
Prefer, as it...